test/session_test/headers_test.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module SessionTest
+  module HeadersTest
+    class ConfigTest < ActiveSupport::TestCase
+      def test_headers_key
+        UserSession.headers_key = "my_headers_key"
+        assert_equal "my_headers_key", UserSession.headers_key
+
+        UserSession.headers_key "user_credentials"
+        assert_equal "user_credentials", UserSession.headers_key
+      end
+
+      def test_single_access_allowed_request_types
+        UserSession.single_access_allowed_request_types = ["my request type"]
+        assert_equal ["my request type"], UserSession.single_access_allowed_request_types
+        UserSession.single_access_allowed_request_types(
+          ["application/rss+xml", "application/atom+xml"]
+        )
+        assert_equal(
+          ["application/rss+xml", "application/atom+xml"],
+          UserSession.single_access_allowed_request_types
+        )
+      end
+    end
```

Comment:
To avoid duplicating `ConfigTest` verbatim, perhaps we should combine `params_test.rb` and `headers_test.rb` and call it `single_access_token_test.rb`.

Reply:
@jaredbeck Note that I don't necessarily like adding a method that isn't really a `test_` method to the class `SessionTest::SingleAccessTokenTest::InstanceMethodsTest` but it was the best way I could see to reduce the duplication, and not hide the actual assertions off in a helper file (which would be far worse for the reader).